runtime/cudaq/utils/details/impls/xtensor_impl.cpp

```diff
@@ -0,0 +1,163 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2024 NVIDIA Corporation & Affiliates.                         *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "cudaq/utils/details/tensor_impl.h"
+#include "cudaq/utils/type_traits.h"
+
+#include <xtensor/xadapt.hpp>
+#include <xtensor/xio.hpp>
+
+#include <fmt/ranges.h>
+
+namespace cudaq {
+
+/// @brief An implementation of tensor_impl using xtensor library
+template <typename Scalar>
```

Comment:
Probably want to use concepts here to force `Scalar` to be a limited set of types.